[U-1396] Add `environment_variables` attribute to `betteruptime_monitor`

dmajda opened the pull request:
The attribute reflects the same attribute in the Monitor API, which was recently added.

dmajda committed:
Make `load` handle `**map[string]string`

dmajda committed:
Add `environment_variables` attribute to `betteruptime_monitor`

The attribute reflects the same attribute in the Monitor API, which was
recently added.

dmajda committed:
Add `environment_variables` to the advanced example

PetrHeinz commented on examples/advanced/main.tf:
```diff
@@ -66,6 +66,9 @@ resource "betteruptime_monitor" "playwright" {
   EOT
   request_timeout   = 60
+  environment_variables = {
+    PASSWORD : "passw0rd"
```
💄 consistent with test cases / other examples

```suggestion
    PASSWORD = "passw0rd"
```

PetrHeinz replied:
💄 It would also be great to use the variable using process.env.PASSWORD in the playwright script

dmajda replied:
> 💄 It would also be great to use the variable using process.env.PASSWORD in the playwright script

I see no "natural" way how to use the password (or similar) on betterstack.com, so I'm thinking of a few options:

  1. Change the example to check something like [https://httpbin.org/basic-auth/{user}/{passwd}](https://httpbin.org/#/Auth/get_basic_auth__user___passwd_). This would demonstrate the intended use, but it's not very realistic and we will unnecessarily spam httpbin.org with requests. This would be similar with any other website.
  2. Change the variable to something like `TITLE` and keep the current example (modulo using the variable). This doesn't show the intended use + also not very realistic.
  3. Keep the example as it is.
  
@PetrHeinz What do you think? Maybe you see some other option I don't?

PetrHeinz replied:
I'd either keep it as is, use move the title to the secret, or use secret `EMAIL = "hello@betterstack.com"` and expect it on the page (it's in the footer).

dmajda replied:
> use secret EMAIL = "hello@betterstack.com" and expect it on the page (it's in the footer)

Thanks, done that.

PetrHeinz commented on internal/provider/resource_monitor.go:
```diff
@@ -401,6 +401,40 @@ var monitorSchema = map[string]*schema.Schema{
 		Computed:    true,
 	},
+	"environment_variables": {
+		Description: "For Playwright monitors, the environment variables that can be used in the scenario.",
```
💄 Feel free to include an example here as well, it's great to see an example straight in the docs. OK if omitted for a reason 👍 

```suggestion
		Description: "For Playwright monitors, the environment variables that can be used in the scenario. Example: `{ "PASSWORD" = "passw0rd" }`.",
```

dmajda replied:
Done. The example was omitted because I copied this from the API docs, where the example wasn't included in all places. This is now also fixed.

PetrHeinz reviewed (approved):
Awesome, thank you 🚀 

Could you please release as `v0.19.3`? 🙏 Just creating a new release via Github UI and letting it generate the description suffices 👍 

---

BTW, avoiding amending commits and force pushes makes the review easier 🙏

dmajda committed:
Add example to `environment_variables` docs

dmajda committed:
Improve advanced example

Change the Playwright environment variable to `EMAIL` and use it in the
scenario. Also use consistent syntax.

dmajda committed:
Bump version to 0.19.3
